src/main/java/com/bootcamp/demo/controller/AccountController.java

```diff
@@ -26,6 +30,37 @@ public AccountController(Service<Account> service) {
         }
     }
 
+    @PutMapping("/editAccount")
+    ResponseEntity<Account> update(@RequestBody AccountDetails accountDetails) throws ControllerException {
+        try {
+            Account searchedAccount = service.findOne(accountDetails.getUsername());
+
+            if (searchedAccount != null) {
+                Account account = new Account(
+                        accountDetails.getName(),
+                        accountDetails.getEmail(),
+                        accountDetails.getUsername(),
+                        searchedAccount.getPassword(),
+                        accountDetails.getPhoneNumber(),
+                        accountDetails.getAddress(),
+                        accountDetails.getDateOfBirth()
+                );
```

Comment:
This is business logic that should be done in the service. If you want to use the accountDetails class, you can add another generic type to the service interface.

Reply:
> This is business logic that should be done in the service. If you want to use the accountDetails class, you can add another generic type to the service interface.

Another idea is to move everything that has to do with transfering data from an account to an accountDetails to the controller and let the service only deal with accounts.